scipy/stats/stats.py

```diff
@@ -2772,11 +2780,15 @@ def spearmanr(a, b=None, axis=0):
 
     """
     a, axisout = _chk_asarray(a, axis)
+    if a.size <= 1:
+        return np.nan, np.nan
     ar = np.apply_along_axis(rankdata, axisout, a)
 
     br = None
     if b is not None:
         b, axisout = _chk_asarray(b, axis)
+        if b.size <= 1:
+            return np.nan, np.nan
```

Comment:
I added the size check here as well. I think it is good for clarity and consistency but it may be redundant because `np.corrcoef` raises an exception unless the sizes `a` and `b` are equal. What do you think @rgommers?